### DISCUSSION TITLE
Find a good / meaningful name

### DISCUSSION THREAD

**Discussion by thebino** [AndroidDev-social/DodoForMastodon#22] [General]
### Description

When talking about a product it is mandatory to have a proper name which can be remembered and linked to its functionality.
Like the community once came up with [Boaty McBoatface](https://en.wikipedia.org/wiki/Boaty_McBoatface) we should be able to find some good names as well.

Let's first collect some ideas and later on we can start a poll for the decision.

**Comment by hector6872** [comment]
Mastodonte 👍🏼 
which is Mastodon translated into Spanish https://www.wordreference.com/es/en/translation.asp?spen=mastodonte

**Comment by waseefakhtar** [comment]
Mammut

Reference:
<img width="1595" alt="Screenshot 2022-11-08 at 8 57 53 PM" src="https://user-images.githubusercontent.com/4093820/200613705-b6e52bb9-61d1-4887-bde4-f6e06ec32947.png">

**Comment by JacquesSmuts** [comment]
Mammut already exists as a Mastodon Android app, though not a very active one atm.
https://github.com/jamiesanson/Mammut

**Comment by JacquesSmuts** [comment]
Woolly

**Comment by evant** [comment]
Kdon (not serious but felt like we needed the obligatory suggestion with a K in it)

**Comment by blvckcoded** [comment]
Multidon
Mastroid
Kotlodon
Mammoth
Manny

Are some really fun/good suggestions as well (Thanks Alex!)

**Comment by evowizz** [comment]
This suggestion is impossible, unfortunately. See: https://github.com/AndroidDev-social/MastodonCompose/discussions/22#discussioncomment-4227739
___
What about **MastodonX** (or **Mastodon X**)?

**Why this name?**
1. In reference to Android**X** (and Jetpack).
2. Because **X** is a cross, and the app is meant to be **cross**-platform.
3. Because it would make it easy to find the app considering it contains _Mastodon_.
4. **X** is also some sort of sign of renewal. I'm not exactly sure how to explain this but for example, Telegram X on iOS replaced the old version of Telegram back in 2018. [Quoting Telegram](https://telegram.org/blog/telegram-x#telegram-x): 
     >The goal of Telegram X is to reinvent Telegram and explore new frontiers in speed, ease of use, quality of animations and all other aspects.

    This project matches the description that Telegram gave to Telegram X back in 2018.
<br/>
What do you think?

**Reply by blvckcoded** [reply]
I love this!!!! 💯

**Reply by evant** [reply]
Calling it 'Mastadon' anything is likely to get you hit with an impersonation suspension. The accepted pattern to show compatibility is by adding 'for Mastodon' at the end. Tusky does this ![Screenshot_20221112-115053.png](https://user-images.githubusercontent.com/628930/201492168-ccb10863-b077-461c-8c5e-eee7a80f57c4.png)

**Reply by blvckcoded** [reply]
Rules are made to be broken! Booooo! 

Okay jokes aside, happy you all are here with all this insight. So many "rules"! Definitely keep the ideas coming. We are a bit away from publishing an MVP and so we def have a good bit of time to come up with a fun catchy name (and not get into trouble 😔)

**Reply by evowizz** [reply]
I thought about it. But I haven't seen anything on https://joinmastodon.org about the name Mastodon. Worst case scenario, one could always contact the company directly and ask if the name can be used.

**Reply by marcin-kozinski** [reply]
So this happened to another Mastodon client: https://floss.social/@mastodos/109386530506679820

**Reply by evowizz** [reply]
Thanks for sharing it @marcin-kozinski. I looked into it and the trademark is filled there.
https://euipo.europa.eu/eSearch/#details/trademarks/018685721

The screenshot in the link you shared comes from the [README](https://github.com/mastodon/mastodon-android/blob/master/README.md) in the GitHub repository of the official Android app.

Too bad... but it's understandable. However, the trademark should definitely be mentioned in more prominent places.

**Comment by digitalbuddha** [comment]
Just as a note. I think Android x == android extensions same as rx is reactivate extension or Javax was for Java. Each was a collection of extensions to a language/framework. Not sure if confusing to call an app somethingX

**Reply by evowizz** [reply]
X can be used for anything really. Extensions, Extended, etc. I already gave Telegram X as an example, but there's also iPhone X (which in this case is "10"). My point is: X can mean various things, it's not limited to "extensions". :)

**Comment by kvithayathil** [comment]
MaXtodon 🤔

**Comment by krysanify** [comment]
- Kandula, from History's 10 Famous Elephants (https://www.history.com/news/10-famous-elephants-from-history)
> Another famed ancient elephant was the trusty companion of King Dutugamunu, the second-century B.C. ruler of Sri Lanka, who famously defeated King Elara, his South Indian rival, to become ruler of the entire island of Ceylon (today’s Sri Lanka). Captured in the forest around the time of Dutugamunu’s birth, the elephant Kandula grew up alongside the young prince. As the royal mount, he performed heroically in the siege of Vijitanagara (161 B.C.), returning to finish breaking down a fortified gate after recovering from having molten pitch poured on his back. According to the Mahavamsa, an ancient Buddhist chronicle, the king rushed to Kandula to administer a salve, exclaiming, “Dear Kandula, I’ll make you the lord of all Ceylon!” Later Kandula was Dutugamunu’s mount in his one-on-one combat when he defeated Elara

- Kailash, the abode of Ganesha, one of many Hinduism deities with an elephant head (https://en.wikipedia.org/wiki/Mount_Kailash)
> Climate change due to [global warming](https://en.wikipedia.org/wiki/Climate_change) is happening three times faster on the [Tibetan Plateau](https://en.wikipedia.org/wiki/Tibetan_Plateau) than anywhere else in the world.[[55]](https://en.wikipedia.org/wiki/Mount_Kailash#cite_note-55) Mount Kailash is located in a mountain range upon the Tibetan Plateau, near the plateau's western edges. According to local observers, the land around Mount Kailash has been growing warmer in recent years; the pilgrimage season isn't as cold as it used to be.

- Kangiten, Japanese Buddhism version of Ganesha (https://en.wikipedia.org/wiki/Kangiten)
> Kangiten enjoys both a positive and negative reception in Japan. On the one hand, he is popularly revered as an extremely efficacious god who grants whatever is asked of him without fail, including impossible wishes. On the other hand, he is considered to be still bound by base passions and desires ([kleshas](https://en.wikipedia.org/wiki/Kleshas_(Buddhism))) and thus is sometimes also regarded as a rather volatile, demanding god who is quick to punish those who have offended him.

For more casual, non cultural appropriation names:
- Booster
- BoostToot
- BoostPub
- Buster
- Busty (my vote)
- Bustly
- TootPub
- TootiePub (also my vote)

**Comment by yogeshpaliyal** [comment]
Mastodroid

**Comment by marcin-kozinski** [comment]
How about a different extinct animal, something not necessarily so gigantic, maybe something cuter that presents some good options for app icon and in-app illustrations, like:

Dodo

**Reply by digitalbuddha** [reply]
I love that it's an extinct bird 😂

**Reply by marcin-kozinski** [reply]
Haha, I didn't even think about it, but that's perfect 😂

**Comment by scottiefalkon** [comment]
Please not MastodonX. The X naming scheme is beyond played out.
Glyph, Kasie (end of African word for communication), Smoke (smoke signals), Unitum (latin for united).

**Comment by blvckcoded** [comment]
I have created a poll in the official slack in the channel #mastodon-client. Please vote for an option there. Thanks.

**Comment by kvithayathil** [comment]
Worth considering (broadly) how it will be pronounced across different languages. Obviously not going to cover every case (or a large number of them), but something to keep in mind.
